### PR TITLE
Materials module including Physically Based Rendering (PBR)

### PR DESCRIPTION
**Pull request by guaje**
This PR adds a new module to manipulate material-like properties of different actors.

The following videos show the effect of different standard (ambient, diffuse and specular levels, and power) and PBR (metallicity and roughness) material properties on the surface actor.

| Standard | PBR |
|---|---|
| <video controls="true" width="350" src="https://user-images.githubusercontent.com/9929496/109573102-614feb80-7abc-11eb-9657-982ce98ad571.mp4"> | <video controls="true" width="350" src="https://user-images.githubusercontent.com/9929496/109573133-6f9e0780-7abc-11eb-900a-f65b4f634a45.mp4"> |

The test cases include unit tests for all the actors in the `actors` module with some comments on special cases where the corresponding material interpolation is not fully functional and cases where the functionality is not supported at all and throws errors.

### CODE REVIEW

File: fury/actor.py [hunk 1]
```diff
@@ -430,6 +439,11 @@ def contour_from_roi(data, affine=None,
     skin_actor.GetProperty().SetColor(color[0], color[1], color[2])
     skin_actor.GetProperty().SetOpacity(opacity)
 
+    if pbr:
+        skin_actor.GetProperty().SetInterpolationToPBR()
+        skin_actor.GetProperty().SetMetallic(metallic)
+        skin_actor.GetProperty().SetRoughness(roughness)
```
Comment (skoudoro) [inline fury/actor.py:445]: It will be good if this block should be a new function with `actor, metallic and roughness` as a parameter

**Review by skoudoro — CHANGES_REQUESTED** [review]
Thanks @guaje for this. 

Overall, the code just needs a redesign. Apart from that, it looks good.

**Review by skoudoro — COMMENTED** [review]
Hi @guaje,

What is the update concerning this PR?

Thank you for your feedback.

File: fury/actor.py (hunk 1 above)
Comment (guaje) [reply fury/actor.py]: I've made the suggested changes and added the tests on the currently implemented actors.

File: fury/material.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        Parameter used to specify how glossy the actor should be. Values must
+        be between 0.0 and 1.0.
+    """
+    prop = actor.GetProperty()
+    prop.SetAmbient(ambient_level)
+    prop.SetDiffuse(diffuse_level)
+    prop.SetSpecular(specular_level)
+    prop.SetSpecularPower(specular_power)
+
+    if interpolation.lower() == 'phong':
```
Comment (skoudoro) [inline fury/material.py:51]: can you add something like 
```
else:
    warnings.warn('Unknow interpolation. Ignoring {} interpolation option')
```

File: fury/tests/test_material.py [hunk 3]
```diff
@@ -0,0 +1,871 @@
...
+
+import numpy as np
+import numpy.testing as npt
+
+
+def test_manifest_pbr(interactive=False):
+    # Setup surface
+    import math
+    import random
+    from scipy.spatial import Delaunay
```
Comment (skoudoro) [inline fury/tests/test_material.py:12]: all this import can be on the top of the file

File: fury/tests/test_material.py [hunk 4]
```diff
@@ -0,0 +1,871 @@
...
+
+    if interactive:
+        window.show(scene)
+
+
+def test_manifest_standard(interactive=False):
+    # Setup surface
+    import math
+    import random
+    from scipy.spatial import Delaunay
```
Comment (skoudoro) [inline fury/tests/test_material.py:429]: duplicate, same comment as above

File: fury/tests/test_material.py [hunk 5]
```diff
@@ -0,0 +1,871 @@
...
+        for j in range(-size, size):
+            fact1 = - math.sin(i) * math.cos(j)
+            fact2 = - math.exp(abs(1 - math.sqrt(i ** 2 + j ** 2) / math.pi))
+            z_coord = -abs(fact1 * fact2)
+            vertices.append([i, j, z_coord])
+    c_arr = np.random.rand(len(vertices), 3)
+    random.shuffle(vertices)
+    vertices = np.array(vertices)
+    tri = Delaunay(vertices[:, [0, 1]])
+    faces = tri.simplices
```
Comment (skoudoro) [inline fury/tests/test_material.py:442]: duplicate, maybe you can create a function above named `generate_data` or similar

**Review by skoudoro — COMMENTED** [review]
Hi @guaje,

Here a quick review. Overall, it looks good. 
Please, check the docstring in `material.py`
Please, try to reduce duplication in `test_material.py`

**Pull request merged**